src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -29,7 +35,12 @@
      * @return collection of user, without duplicates
      */
     public List<User> createUsers(final List<String> emails) {
-        return null;
+        List<User> users = emails.stream()
+                .filter(email-> email!=null && !email.isEmpty())
+                .distinct()
+                .map(p -> new User(p))
```

Comment:
You could've used method references here (like `User::new`), but plain lambda is also fine.